### ISSUE TITLE
Make 'upgrade' command work with partial names

### ISSUE THREAD

**Issue by Nino-kun** [microsoft/winget-cli#2597]
### Description of the new feature / enhancement

Due to bad vision I have to use large console fonts. Because of that winget table with list of available upgrades don't have packages full names or ids
`Name                  Id                    Version       Available     Source
------------------------------------------------------------------------------
GPL Ghostscript       ArtifexSoftware.Ghos… 9.27          10.00.0       winget
Wireshark 3.6.3 64-b… WiresharkFoundation.… 3.6.3         4.0.0.0       winget
Microsoft Visual C++… Microsoft.VCRedist.2… 14.29.30139.0 14.34.31823.3 winget
Microsoft Visual Stu… Microsoft.VisualStud… 1.70.1        1.72.1        winget
Microsoft Visual C++… Microsoft.VCRedist.2… 14.29.30139.0 14.34.31823.3 winget
eM Client             eMClient.eMClient     9.0.1361.0    9.1.2109.0    winget`
If I don't want to upgrade everything, it become really difficult to find name, or id, or moniker of packages for upgrade. Running "_winget upgrade 'Microsoft Visual C'_" or "_winget upgrade --name 'Microsoft Visual C'_" returned "_No installed package found matching input criteria._" It took me more than 5 minutes to find what 'Microsoft Visual C++…' is actually Microsoft Visual C++ 2015-2022 Redistributable (x64) and (x86) and only after that I was able to upgrade this packages.

### Proposed technical implementation details

Make winget _upgrade_ command accept part of name or id as input, like _list_ command. Or make it accept wildcards. Or add option for display result not only as table, but as list as well.

**Comment by Trenly** [comment]
You may want to take a look at adding [Tab Completion](https://github.com/microsoft/winget-cli/blob/master/doc/Completion.md) to your profile. It will allow you to enter a partial name / id and then you can use tab to cycle through the available options

**Comment by Nino-kun** [comment]
Thank you for suggestion. It become much easier to find full package name.

**Comment by denelon** [comment]
@Nino-kun we have done some work on accessibility and screen readers. I don't know how great the experience is, but that may be another option. 

Please feel free to submit any suggestions that would make this product easier for you to use. I care deeply about the product, and I want to make sure everyone has a great experience.

We are also actively working on a native PowerShell implementation that might also offer a better experience since the "output" can be formatted in a better way, and there will be options that do not truncate the text based on the number of characters the terminal or shell is able to output.

**Comment by cjwijtmans** [comment]
Or instead of tables one could display it as a simple compact string. Such as `<id>:<version>[<source>] -> <upgrade version>` on each line. Winget info <id> to get the description and display name.

**Comment by Trenly** [comment]
[Policy] Command-Upgrade

**Comment by Thaina** [comment]
![Image](https://github.com/user-attachments/assets/be1b7328-67d5-4fb8-89b2-995b1c98533f)

I want to use `winget upgrade Microsoft.DotNet.SDK.* Python.Python.3.*` in this case

**Comment by stevenwdv** [comment]
Is this a duplicate of #2014?
